packages/common/lib/xml-ns-key.js

```diff
@@ -0,0 +1,38 @@
+// The xml parser takes care of validating the attribute name.
+// If the user started the attribute name with "xmlns:" we can assume that
+// they meant for it to be an xml namespace attribute.
+// xmlns attributes explicitly can't contain ":" after the "xmlns:" part.
+const namespaceRegex = /^xmlns(?<prefixWithColon>:(?<prefix>[^:]*))?$/;
+
+function isXMLNamespaceKey(key, includeEmptyPrefix) {
```

Comment:
use simple JSDocs to give IDE hints for types: e.g:
- https://github.com/SAP/xml-tools/pull/173/files#diff-7c5464133cf9296983157483270a9fd4R22-R27

Reply:
The link doesn't work

Reply:
Attempt \#2 https://github.com/SAP/xml-tools/blob/master/packages/ast/lib/build-ast.js#L18-L23

Reply:
Added